float64 v
float64 omega

---
float64 omega_l
float64 omega_r
